The error seen at this point, only on Windows at https://github.com/Electron100/butane/actions/runs/19286195451/job/55147496873?pr=420 is

```
cargo +stable nextest run --all-features
  shell: C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
  env:
    CARGO_TERM_COLOR: always
    PQ_LIB_DIR: C:\Program Files\PostgreSQL\12\lib
    BUTANE_PG_CONNSTR: host=localhost user=postgres *** sslmode=disable port=5432
    CARGO_INCREMENTAL: 0
    CARGO_PROFILE_DEV_DEBUG: 0
    RUST_BACKTRACE: short
    RUSTFLAGS: -D warnings
    CARGO_UNSTABLE_SPARSE_REGISTRY: true
    CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
    CACHE_ON_FAILURE: true
   Compiling serde_core v1.0.228
   Compiling git2 v0.20.2
   Compiling phf_generator v0.13.1
   Compiling object v0.32.2
   Compiling phf_macros v0.13.1
   Compiling phf v0.13.1
   Compiling tracing-subscriber v0.3.20
   Compiling serde v1.0.228
   Compiling ar_archive_writer v0.2.0
   Compiling built v0.7.7
   Compiling chrono v0.4.42
   Compiling serde_json v1.0.145
   Compiling psm v0.1.28
   Compiling turso_core v0.3.0
   Compiling stacker v0.1.22
   Compiling uuid v1.18.1
   Compiling recursive v0.1.1
   Compiling sqlparser v0.59.0
   Compiling postgres-types v0.2.11
   Compiling turso_ext v0.3.0
   Compiling tokio-postgres v0.7.15
   Compiling fake v4.4.0
   Compiling rusqlite v0.37.0
   Compiling postgres-native-tls v0.5.2
   Compiling desynt v0.1.0
   Compiling url v2.5.7
   Compiling libsqlite3-sys v0.35.0
   Compiling serde_variant v0.1.3
   Compiling camino v1.2.1
   Compiling cargo-platform v0.3.1
   Compiling semver v1.0.27
   Compiling cargo_metadata v0.23.0
   Compiling butane_tests v0.8.1 (D:\a\butane\butane\butane\tests)
warning: butane_tests@0.8.1: Deleting .butane directory
   Compiling atomicwrites v0.4.4
   Compiling expectorate v1.2.0
   Compiling example v0.1.0 (D:\a\butane\butane\example)
   Compiling test-log v0.2.18
   Compiling serde_spanned v1.0.3
   Compiling toml_datetime v0.7.3
   Compiling bstr v1.12.1
   Compiling winapi-util v0.1.11
   Compiling toml v0.9.8
   Compiling assert_cmd v2.0.17
   Compiling termcolor v1.4.1
   Compiling trybuild v1.0.114
   Compiling butane_test_macros v0.8.1 (D:\a\butane\butane\butane_test_macros)
   Compiling turso v0.3.0
   Compiling butane_core v0.8.1 (D:\a\butane\butane\butane_core)
   Compiling butane_test_helper v0.8.1 (D:\a\butane\butane\butane_test_helper)
   Compiling butane_codegen v0.8.1 (D:\a\butane\butane\butane_codegen)
error: linking with `link.exe` failed: exit code: 1181
  |
  = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.44.35207\\bin\\HostX64\\x64\\link.exe" "/DEF:C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcoGfUxF\\lib.def"
....
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: LINK : fatal error LNK1181: cannot open input file 'sqlite3.lib'␍

error: could not compile `butane_codegen` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: command `'C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin\cargo.exe' test --no-run --message-format json-render-diagnostics --all-features` exited with code 101
```

This is typical of when sqlite feature has been enabled, but sqlite-bundled has not.  This doesnt make a lot of sense to me because if building at the workspace level `--all-features` should be enabling sqlite-bundled also.
And if building only at the butane_codegen level , there is no sqlite deps:

```
butane_codegen % cargo tree --invert rusqlite
error: package ID specification `rusqlite` did not match any packages
```

--- a/.config/nextest.toml
+++ b/.config/nextest.toml
@@ -8,10 +8,6 @@
 # Default filter runs all tests on non-Windows platforms
 default-filter = 'all()'
 
-[profile.ci]
-# Default filter runs all tests on non-Windows platforms
-default-filter = 'all()'
-
 # Platform-specific override to exclude butane_codegen on Windows
 [[profile.default.overrides]]
 platform = "cfg(windows)"
@@ -23,12 +19,48 @@ platform = "cfg(macos)"
 filter = 'test(cleanup_on_drop)'
 retries = 3
 
-[[profile.ci.overrides]]
-platform = "cfg(windows)"
-default-filter = 'not package(butane_codegen)'
+# Test priority configuration - run tests in order of importance
+[[profile.default.overrides]]
+filter = 'package(butane_test_helper)'
+priority = 100
 
-# Retry flaky test on macOS
-[[profile.ci.overrides]]
-platform = "cfg(macos)"
-filter = 'test(cleanup_on_drop)'
-retries = 3
+[[profile.default.overrides]]
+filter = 'package(butane_core)'
+priority = 80
+
+[[profile.default.overrides]]
+filter = 'package(butane_cli)'
+priority = 60
+
+[[profile.default.overrides]]
+filter = 'package(butane) and not package(butane_cli) and not package(butane_core) and not package(butane_codegen) and not package(butane_test_helper) and not package(butane_test_macros)'
+priority = 40
+
+[[profile.default.overrides]]
+filter = 'package(trybuild_tests)'
+priority = 20
+
+# Example packages - all at same priority level
+[[profile.default.overrides]]
+filter = 'package(example)'
+priority = 10
+
+[[profile.default.overrides]]
+filter = 'package(custom_pg)'
+priority = 10
+
+[[profile.default.overrides]]
+filter = 'package(newtype)'
+priority = 10
+
+[[profile.default.overrides]]
+filter = 'package(getting_started)'
+priority = 10
+
+[[profile.default.overrides]]
+filter = 'package(getting_started_async)'
+priority = 10
+
+[[profile.default.overrides]]
+filter = 'package(reserved-words)'
+priority = 10

--- a/.config/nextest.toml
+++ b/.config/nextest.toml
@@ -17,6 +17,18 @@ default-filter = 'all()'
 platform = "cfg(windows)"
 default-filter = 'not package(butane_codegen)'
 
+# Retry flaky test on macOS
+[[profile.default.overrides]]
+platform = "cfg(macos)"
+filter = 'test(cleanup_on_drop)'
+retries = 3
+
 [[profile.ci.overrides]]
 platform = "cfg(windows)"
 default-filter = 'not package(butane_codegen)'
+
+# Retry flaky test on macOS
+[[profile.ci.overrides]]
+platform = "cfg(macos)"
+filter = 'test(cleanup_on_drop)'
+retries = 3

--- a/.config/nextest.toml
+++ b/.config/nextest.toml
@@ -1,0 +1,22 @@
+# Skip testing the butane_codegen workspace member on Windows only.
+# It is a proc-macro crate that has test = false in its Cargo.toml
+# due to issues with workspace-level testing
+# See [issue #419](https://github.com/Electron100/butane/issues/419)
+# and PR [#420](https://github.com/Electron100/butane/pull/420).
+
+[profile.default]
+# Default filter runs all tests on non-Windows platforms
+default-filter = 'all()'
+
+[profile.ci]
+# Default filter runs all tests on non-Windows platforms
+default-filter = 'all()'
+
+# Platform-specific override to exclude butane_codegen on Windows
+[[profile.default.overrides]]
+platform = "cfg(windows)"
+default-filter = 'not package(butane_codegen)'
+
+[[profile.ci.overrides]]
+platform = "cfg(windows)"
+default-filter = 'not package(butane_codegen)'

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -117,10 +117,15 @@ jobs:
       run: cargo deny check all
     - name: Test with mise tools
       if: runner.os != 'Windows'
-      run: mise exec ephemeral-postgres -- cargo +stable nextest run --all-features
+      run: mise exec ephemeral-postgres -- cargo +stable nextest run --all-features --no-fail-fast
     - name: Test without mise
       if: runner.os == 'Windows'
-      run: cargo +stable nextest run --all-features
+      run: cargo +stable nextest run --all-features --no-fail-fast
+
+    - name: Test doctests
+      run: |
+        cargo +stable test -p butane_codegen --doc
+        cargo +stable test -p butane --doc --features async
 
     - name: Check example migrations have been updated
       run: |

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -120,7 +120,7 @@ jobs:
       run: mise exec ephemeral-postgres -- cargo +stable nextest run --all-features --no-fail-fast
     - name: Test without mise
       if: runner.os == 'Windows'
-      run: cargo +stable nextest run --all-features --no-fail-fast
+      run: cargo +stable nextest run --all-features --no-fail-fast --verbose
 
     - name: Test doctests
       run: |

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -97,12 +97,20 @@ It is also recommended to use cargo-nextest which provides many benefits, especi
 cargo nextest run
 ```
 
+Note cargo-nextest does not support doctests, [yet](https://github.com/nextest-rs/nextest/issues/16).
+
+To check those:
+
+```bash
+cargo test -p butane --doc --features async
+cargo test -p butane_codegen --doc  # all doctest are ignored
+```
+
 Run tests for a specific package:
 
 ```bash
-cargo test -p butane_core
-cargo test -p butane  # doctests only
-cargo test -p butane_tests
+cargo test -p butane_core --all-features
+cargo test -p butane_tests --all-features
 cargo test -p trybuild_tests
 ```
 

